app/src/main/java/org/akvo/flow/activity/AddUserActivity.java

```diff
@@ -61,6 +63,10 @@ public void onClick(View v) {
                 db.savePreference(ConstantUtil.DEVICE_IDENT_KEY, deviceId);
                 db.close();
 
+                // Trigger the SurveyDownload Service, in order to make
+                // a backend connection with the new Device ID
```

Comment:
I guess I would be in favour of moving it there, as it will clarify the intent and structure of the code.